docs/staking-entry-queue/dd.md

```diff
@@ -0,0 +1,136 @@
+- Approvers:
+  - @LHerskind
+  - @Maddiaa0
+  - @aminsammara
+- Target DD Approval Date: 2025-06-10
+- Target Project Delivery Date: 2025-06-15
+
+## Executive Summary
+
+Requirement: Mitigate the risk of a large influx of byzantine stakers which could threaten the stability/security of the network.
+
+We accomplish this by introducing a staking entry queue.
+
+That is, when a staker deposits into the aztec network, they are added to a queue on L1.
+
+## Timeline
+
+Outline the timeline for the project. E.g.
+
+- Build L1 queue: 1 day
+- Update typescript: 1 day
+- Review/polish: 1 day
+
+Total: 3 days
+
+## Introduction
+
+It is not good for it to be possible for a large number of validators to (near) instantly start participating in block production.
+
+This can result in, for example, where a significant portion of the validator set is malicious or faulty.
+
+This is a particular concern at the outset of a network where it is possible for a large influx to occur in the absence of any flow control mechanism.
+
+This can be problematic for the following reasons:
+
+- It is possible that the nodes are improperly configured (accidentally or otherwise), and are not actually able to participate in block production
+- It is possible that the p2p network will not form a well connected mesh, hindering throughput
+
+It is better to allow a more gradual transition, rather than shock the system.
+
+## Interface
+
+Users are the would-be and current validators of the aztec network.
+
+Currently, users are able to go directly to rollup instances, and call
+
+```solidity
+  function deposit(address _attester, address _withdrawer, bool _onCanonical) external;
+```
+
+This interface will remain _unchanged_, but the semantics will be that they are now "enqueued".
+
+Users _will_ need to call `flushEntryQueue` though to move into the active state of validating.
+
+```solidity
+  function flushEntryQueue() external;
+```
+
+## Implementation
+
+When would-be validators call `deposit`, they will enter a queue maintained by the rollup instance.
+
+Validators in the queue do not participate in block production, and will not be sampled during committee selection.
+
+When _anyone_ calls `flushEntryQueue` for a given epoch `e`, then up to `N` validators will be dequeued and added to the validator set, and be eligible to serve on the committee for epoch `e`.
+
+_NOTE_: we will restrict it such that `flushEntryQueue` may only be called once for epoch `e`, and only during epoch `e-1`.
+
+_NOTE_: we currently setup epochs 1 full epoch in advance, so in the first slot in epoch `e-1` we can setup epoch `e`. So in most cases, it will be that the queue is drained during epoch `e-1`, and the new validators may appear in the committee for epoch `e`, and that will be know effectively at the outset of epoch `e-1`. If no one is calling `setupEpoch` though for multiple epochs, there could be a build up of new validators that could be eligible, all in the same epoch.
```

Comment:
As mentioned earlier, they won't be in the committee for `e` because that is controlled separately, but they will be in the validator set. Think I would try to avoid all of these `e-1` and `e` notions